software/generic/ControlEffect.cpp

```diff
@@ -0,0 +1,21 @@
+#include "ControlEffect.hpp"
+
+static inline uint8_t min(uint8_t a, uint8_t b) {
```

Comment:
Is this needed?

Reply:
`min` does not seem to be defined here, is there a header I need to include? I already tried `Arduino.h`.

Reply:
Types.hpp undefs the Arduino macro for this (boo to macros). Can you use std::min?

Reply:
Oh, hold on, I'm not using it in this version (I implemented software smoothing to make packet loss less obvious but it didn't help enough). I'll remove this altogether an try using `std::min` in the future.